cluster-autoscaler/loop/trigger.go

```diff
@@ -74,16 +97,23 @@ func (t *LoopTrigger) Wait(lastRun time.Time) {
 		default:
 			klog.Infof("Autoscaler loop triggered immediately after a productive iteration")
 		}
-		return
-	}
-
-	// Unschedulable pod triggers autoscaling immediately.
-	select {
-	case <-time.After(t.scanInterval):
-		klog.Infof("Autoscaler loop triggered by a %v timer", t.scanInterval)
-	case <-t.podObserver.unschedulablePodChan:
-		klog.Info("Autoscaler loop triggered by unschedulable pod appearing")
+	} else if t.provisioningRequestProcessTimeGetter != nil && !t.provisioningRequestProcessTimeGetter.LastProvisioningRequestProcessTime().Before(lastRun) {
```

Comment:
nit: can you add a small wrapper func "provReqWasProcessed" (or some other name) and update the first "if" to:
```
if !t.scalingTimesGetter.LastScaleUpTime().Before(lastRun) ||
		!t.scalingTimesGetter.LastScaleDownDeleteTime().Before(lastRun) ||
       t.provReqWasProcessed(lastRun)
```

This would help to avoid repeating the "if" branch code content.

Reply:
Since we are triggering next loop immediately even after a failed ProvisioningRequest, I feel that logging "Autoscaler loop triggered immediately after a productive iteration" is counter-intuitive.

Reply:
Added a  function to avoid replicating if statement body